### ISSUE TITLE
operator: Allow providing TLS certificate for database connection

### ISSUE THREAD

**Issue by nielspardon** [keycloak/keycloak#15986]
### Description

When using a database with a custom TLS certificate then I currently could not figure out a way to provide the TLS certificate via the Keycloak operator that should be used for the JDBC connection. It would be great if this information could be provided.

### Motivation

Managed databases like IBM Cloud Databases for PostgreSQL deploy the database with a custom TLS certificate. So using such a managed database with Keycloak requires configuring a TLS certificate for the JDBC connection.

### Details

Currently getting:
```
Could not open SSL root certificate file //.postgresql/root.crt.
```

**Comment by shawkins** [comment]
Covered by #12722

**Comment by vmuzikar** [comment]
@nielspardon Or is this about supporting mutual TLS?

**Comment by nielspardon** [comment]
It has been quite a while since I opened this issue. Back then I could not figure out how to provide the trust store for use with the database connection through the Keycloak CR other than using a custom pod template if I remember correctly. For me it looked like all the other options were mainly for ingress TLS keystores / truststores. 

I would have loved to provide a bit more detail but I have moved on to working on other topics meanwhile.
